### PR TITLE
Feature/gridview-min-max-widths

### PR DESCRIPTION
**Pull request by koal44**
Add `MinWidth` and `MaxWidth` support for `GridView`
## Pull request type

- [x] Feature

## Info

This feature is a branch off of PR #979. It adds min/max width support for `GridView`. It uses a fair bit of reflection to maintain support for dragging column headers.

- Extends `GridViewColumn` to have dependency properties `MaxWidth` and `MinWidth`
- Overrides `ArrangeOverride()` on `GridViewHeaderRowPresenter` (header row) and `GridViewRowPresenter` (data rows)
- Update `ListView` and `ListViewItem` to use the extended classes
- Update the GalleryApp to showcase min/max widths

### CODE REVIEW

File: src/Wpf.Ui/Controls/ListView/GridViewColumn.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT was not distributed with this file, You can obtain one at https://opensource.org/licenses/MIT.
+// Copyright (C) Leszek Pomianowski and WPF UI Contributors.
+// All Rights Reserved.
+
+using System.Reflection;
+
+namespace Wpf.Ui.Controls;
+
+public class GridViewColumn : System.Windows.Controls.GridViewColumn
```
Comment (pomianowski) [inline src/Wpf.Ui/Controls/ListView/GridViewColumn.cs:10]: Is the **Controls/ListView** directory the right place for these classes? Maybe **Controls/GridView** would be more appropriate

File: src/Wpf.Ui/Controls/ListView/GridViewColumn.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        if (d is not GridViewColumn self)
+        {
+            return;
+        }
+
+        self.OnMinWidthChanged(e);
+    }
+
+    protected virtual void OnMinWidthChanged(DependencyPropertyChangedEventArgs e)
```
Comment (pomianowski) [inline src/Wpf.Ui/Controls/ListView/GridViewColumn.cs:41]: Missing docs for public api

File: src/Wpf.Ui/Controls/ListView/ListView.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+namespace Wpf.Ui.Controls;
+
+public class ListView : System.Windows.Controls.ListView
```
Comment (pomianowski) [inline src/Wpf.Ui/Controls/ListView/ListView.cs:3]: Missing docs for public control

File: src/Wpf.Ui/Controls/ListView/GridViewRowPresenter.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
+using System.Windows.Controls;
+
+namespace Wpf.Ui.Controls;
+
+public class GridViewRowPresenter : System.Windows.Controls.GridViewRowPresenter
```
Comment (pomianowski) [inline src/Wpf.Ui/Controls/ListView/GridViewRowPresenter.cs:5]: Missing docs for public control

File: src/Wpf.Ui/Controls/ListView/GridViewHeaderRowPresenter.cs [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+// If a copy of the MIT was not distributed with this file, You can obtain one at https://opensource.org/licenses/MIT.
+// Copyright (C) Leszek Pomianowski and WPF UI Contributors.
+// All Rights Reserved.
+
+using System.Reflection;
+using System.Windows.Controls;
+
+namespace Wpf.Ui.Controls;
+
+public class GridViewHeaderRowPresenter : System.Windows.Controls.GridViewHeaderRowPresenter
```
Comment (pomianowski) [inline src/Wpf.Ui/Controls/ListView/GridViewHeaderRowPresenter.cs:11]: Missing docs for public control

**Review by pomianowski — CHANGES_REQUESTED** [review]

**Comment by koal44** [comment]
I've added documentation and separated `GridView` related components into a Controls/GridView folder.

The `ArrangeOverride()` methods on RowPresenter classes was simplified and there's only one reflected field now. Additionally XAML styling is only on Wpf.Ui custom controls, so this library won't interfere with how vanilla `ListView` operates. Instead, to use the Wpf.Ui library's `ListView` styles, one would need to use all the ui custom classes.

TODO: restyle `GridViewColumnHeader.xaml` to support column resizing (logic should already be baked into the control).

**Review by pomianowski — APPROVED** [review]

**Comment by koal44** [comment]
Added drag resizing support.

gridview-resizing

**Review by pomianowski — APPROVED** [review]

**Pull request merged**